### ISSUE TITLE
Downloading an image from gallery view: owncloud crashes after the download step.

### ISSUE THREAD

**Issue by Wikinaut** [owncloud/android#1214]
How to reproduce.

I open a folder "photos" which only has image files (jpg, png). Not a list of files, but a gallery is then shown, three column layout. In my case, the directory contains 12 photos and I see a gallery of 3 columns and 4 rows.

When I click onto one photo. the download to my android device starts (and works), then while apparently trying to display the image the automatic opening fails and owncloud app crashes "Ledier wurde owncloud beendet". 

This happens on two different devices.

Noticed with versions
* Android app version official release 1.8.0 (from Google Play)
* current master (23.10.2015) 2702951056c4b2dc0e6c085088a467601b331c2d

**Comment by AndyScherzinger** [comment]
@Wikinaut can you provide a stace trace from the logcat console? Thius would help identifying the source of the crash

**Comment by Wikinaut** [comment]
@AndyScherzinger Honestly, I don't know, how I can do this. Pls. send me a link to a page describing this,

**Comment by AndyScherzinger** [comment]
Actually I do not have a link. When you connect your phone via an USB cable to your computer and open up Android studio or Eclipse adt you have the logcat at the bottom of your ide showing you the phones log :) 
Hope that helps.

**Comment by tobiasKaminsky** [comment]
http://forum.xda-developers.com/showthread.php?t=1726238
Maybe this helps?

**Comment by LukeOwncloud** [comment]
@AndyScherzinger and @tobiasKaminsky, did you ever use an `UncaughtExceptionHandler` on Android? I always wanted to suggest that we use one but since I never encountered any report of an app crash, I thought it doesn't matter. No it does... -> http://www.intertech.com/Blog/android-handling-the-unexpected/

**Comment by AndyScherzinger** [comment]
@LukeOwncloud not that I remember, but we used bugsense in Andlytics to catch/capture errors and send them to the server. With ownCloud I leave this up to @davivel @rperezb @masensio since like you mentioned I will probably never see an app crash reports being a contributor but this might be something of interest for the team operating the app :)

**Comment by tobiasKaminsky** [comment]
@LukeOwncloud Thank you for this very useful link.
I am considering to implement this in the beta version with the possibility to save the output.
With that every user should be able to post the output to github.

**Comment by Wikinaut** [comment]
I am listening here, but being more interested, if anyone else can confirm my observation (the issue subject).

**Comment by tobiasKaminsky** [comment]
No, I think we need an output of logCat or something similar.
Everything else would be just guessing...
Do you have enough space left on the device?

**Comment by Wikinaut** [comment]
As I told you, I want to see, if anyone else has this problem. I cannot produce logCat output.

**Comment by Wikinaut** [comment]
@tobiasKaminsky Try it on your device.

**Comment by tobiasKaminsky** [comment]
I have and on an emulator with Nexus 5.
I doubt that this is a common problem as otherwise many user would have it, especially if it would be this easy to reproduce.

Can you try it with demo.owncloud.org and test both for user /password?

**Comment by Wikinaut** [comment]
Do you have the problem, or not?

**Comment by tobiasKaminsky** [comment]
No, tried on my Nexus 4 and with emulator on Nexus 5.

**Comment by LukeOwncloud** [comment]
@tobiasKaminsky I can recommend this code: http://stackoverflow.com/questions/23486627/catching-error-and-user-information#answer-23486834
For an activity displaying and sending mail this should be a good starting point: https://github.com/owncloud/android/blob/master/src/com/owncloud/android/ui/activity/LogHistoryActivity.java
I'd love to do it right now, but I still do not have the Android dev tools installed.

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger @LukeOwncloud can you confirm this issue?
@Wikinaut please try it with the demo account: demo.owncloud.org and test both for user /password?

**Comment by LukeOwncloud** [comment]
I do not have version 1.8 installed. I tried to reproduce this issue with the beta version, no success, i.e. no crash.

**Comment by Wikinaut** [comment]
@tobiasKaminsky I tested the three test photos from the demo account, on my two devices ==> test okay, no crash.I retested with my server, which I recently upgraded to 8.2.0 ==> still some images crash the client immediately after the download (the download as such works).

Can you upload further png and gif images to test/photo ? I could do this too. Pls. let me know if you want me to upload to there.

**Comment by tobiasKaminsky** [comment]
The demo.owncloud.org account is periodically recreated.
So you can do anything you want there.
But what would be the benefit of this?
I suspect that it has something to do with
- your server
- the images your are downloading

**Comment by Wikinaut** [comment]
In which cases is a directory rendered as gallery in the client? Can I force the file view for a directory which only contains images ?

**Comment by tobiasKaminsky** [comment]
All files in a folder must be images. So if you add one ore more txt files it will get displayed as list.
You can join IRC: https://webchat.freenode.net/ 
The channel is owncloud-android

**Comment by Wikinaut** [comment]
@tobiasKaminsky I uploaded one new photo ```20111106_Keine_Zeit, den Zaun_zu_reparieren.png``` to demo-oc. Please can you try to download it to your android ? It crashes after download.

**Comment by tobiasKaminsky** [comment]
@Wikinaut it seems as it has been deleted.
Can you join irc? Then we can "talk" more instant

**Comment by tobiasKaminsky** [comment]
@Wikinaut Please install this new beta version:
https://github.com/owncloud/android/blob/beta/apks/latest.apk
It has a new option: After a crash it collects the stacktrace, you can view it and then send it via mail or save it to file.
If the crash occurs again you can then easily post it here

**Comment by Wikinaut** [comment]
@tobiasKaminsky the latest.apk, is this bba296472602f725df7ab4bb2c4a282b40c737ad or something else ? I build bba296472602f725df7ab4bb2c4a282b40c737ad (this has the same problem).

**Comment by Wikinaut** [comment]
die Beta geht nicht richtig, schau selbst: no preview image auf dem Server. 
Missing in the client is the "sync" icon. I only see the sort icon and the file/gallery icon in the upper right corner.

**Comment by Wikinaut** [comment]
Download of my file zaun.png however works now without a crash. I do not see the a preview icon - there is only the image placeholder. Same on the server (web-browser-GUI): only placeholder icon.

**Comment by Wikinaut** [comment]
placeholder icon gone after deletion (on the server), and re-upload

**Comment by Wikinaut** [comment]
but NO PREVIEW icon in the gallery view on the device.

**Comment by tobiasKaminsky** [comment]
Is this only happening for this image?
Can you please share it with me:
https://owncloud.tobiaskaminsky.de/index.php/s/kZ41pl8UPDieBQM
password: owncloud

Regarding the missing sync icon: If this is a bug/regression in the beta version, please open a new issue and label it with "beta".

**Comment by LukeOwncloud** [comment]
@tobiasKaminsky Your link does not work.

**Comment by Wikinaut** [comment]
@tobiasKaminsky chat please

**Comment by Wikinaut** [comment]
@LukasReschke What's the difference between the  latest.apk and what I built from master branch yesterday, the two differ substantially.

**Comment by Wikinaut** [comment]
@LukasReschke oops, I saw, you have a branch ```beta```.

**Comment by Wikinaut** [comment]
@tobiasKaminsky @LukasReschke I built the beta version here (works). I uploaded zaun.png twice to demo-oc, thumbnails were generated & shown on both the server (browser/gui) and the android client. Download worked w/o problems.

So I am closing this issue now. Will reopen it, if necessary. Let me know your e-mail address, then I will send you the image for further tests - perhaps it's a special png-format problem.
